include/deal.II/grid/tria.h

```diff
@@ -24,13 +24,19 @@
 #include <deal.II/base/geometry_info.h>
 #include <deal.II/base/iterator_range.h>
 #include <deal.II/base/std_cxx11/function.h>
+#include <deal.II/base/std_cxx11/unique_ptr.h>
 #include <deal.II/grid/tria_iterator_selector.h>
 
 // Ignore deprecation warnings for auto_ptr.
 DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
 #include <boost/signals2.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/serialization/map.hpp>
+#ifdef DEAL_II_WITH_CXX11
+#  include <boost/serialization/unique_ptr.hpp>
+#else
+#  include <boost/serialization/scoped_ptr.hpp>
```

Comment:
Well, strictly speaking it indeed unnecessary because we already `#include` it from `base/std_cxx11/unique_ptr.hpp`. But that's because I needed to work around things there. I think I'd like to keep it here for completeness in case I come up with a better approach there.